coralproject/talk
# Adjusted stream to preserve comment view state

## What does this PR do?

The stream adjusts the way that comments in a stream are retrieved to ensure that the display is preserved. Specifically, it ensures that the only comments ever displayed are those with a status of `null` (new comment in post-mod mode), and `accepted` (approved comment in pre-mod and post-mod mode).

## How do I test this PR?

1. Run tests
2. Create some comments on a stream in post-mod mode, flag + reject a few, note the comments displayed
3. Switch stream to pre-mod mode
4. Verify that all comments from step 2 are still there
5. Create some comments + approve them, note the comments displayed
6. Switch stream to post-mod mode
7. Verify that all comments from step 5 are still there

Relevant Pivotal Story: https://www.pivotaltracker.com/story/show/137368001
